common/src/main/java/io/netty/util/HashedWheelTimer.java

```diff
@@ -419,6 +419,11 @@ public Timeout newTimeout(TimerTask task, long delay, TimeUnit unit) {
         // Add the timeout to the timeout queue which will be processed on the next tick.
         // During processing all the queued HashedWheelTimeouts will be added to the correct HashedWheelBucket.
         long deadline = System.nanoTime() + unit.toNanos(delay) - startTime;
+
+        // Guard against overflow.
+        if (deadline < 0) {
```

Comment:
@normanmaurer what about the case when someone intentionally passes a negative value into delay?

Reply:
@johnou hmm good question. I am note sure this is expected but let me add a check as it was "allowed" before.